### PR TITLE
JP-3664: Numpy 2.0 incompatibility with existing use of newbyteorder

### PR DESCRIPTION
**Pull request by tapastro**
Resolves [JP-3664](https://jira.stsci.edu/browse/JP-3664)

Partially closes [jwst/#8580](https://github.com/spacetelescope/jwst/issues/8580)

This PR addresses a change in behavior for numpy 2.0: 

> arr.newbyteorder('S').byteswap(inplace=True)
> \>\>\> AttributeError: `newbyteorder` was removed from the ndarray class in NumPy 2.0. Use arr.view(arr.dtype.newbyteorder(order))' instead.

This PR implements the dtype method.

**Checklist**

- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)

### CODE REVIEW

**Review by kmacdonald-stsci — APPROVED** [review]
We already have many `STCAL` PR's needing approval and merging.  I prefer to wait on those being merged before this one is merged, just in case.  I'll approve this, but would like to wait on the merge.

I have not done any investigation into using Numpy 2.0 with ramp fitting, including the C-extension.  Is this the only change that is required or will a more thorough investigation be required?

File: CHANGES.rst
Comment (zacharyburnett) [inline CHANGES.rst:18]:
```suggestion
```
#284 switched change log handling to `towncrier`, so this change log entry should be a file in `changes/`:
```shell
echo "Implement byteorder swap method that is forward-compatible with numpy 2.0 in jwst ramp_fitting." > changes/282.bugfix.rst
```

(new PRs will have these instructions in the checklist)

**Review by braingram — APPROVED** [review]

**Pull request merged**
